iree/samples/variables_and_state/variables_and_state.ipynb

```diff
@@ -0,0 +1,471 @@
+            "\n",
+            " ```\n",
+            "\n",
+            "Wrote MLIR to path '/tmp/iree/colab_artifacts/counter.mlir'\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "WCiRV6KRh3iA"
+      },
+      "source": [
+        "## Test the imported program\n",
+        "\n",
+        "_Note: you can stop after each step and use intermediate outputs with other tools outside of Colab._\n",
+        "\n",
+        "* _The \"imported MLIR\" can be used by IREE's generic compiler tools that run on Linux, macOS, and Windows_\n",
+        "* _The \"flatbuffer blob\" can be saved and used by runtime applications_\n",
+        "\n",
+        "_The specific point at which you switch from Python to native tools will depend on your project._"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "GF0dzDsbaP2w",
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "outputId": "066014a5-134f-45ce-c390-fda4b7f0aa02"
+      },
+      "source": [
+        "#@title Compile the imported MLIR further into an IREE VM bytecode module\n",
+        "\n",
+        "flatbuffer_blob = compile_str(compiler_module, target_backends=[\"vmvx\"])\n",
+        "\n",
+        "# Save the imported MLIR to disk.\n",
+        "flatbuffer_path = os.path.join(ARTIFACTS_DIR, \"counter_vmvx.vmfb\")\n",
+        "with open(flatbuffer_path, \"wb\") as output_file:\n",
+        "  output_file.write(flatbuffer_blob)\n",
+        "print(f\"Wrote .vmfb to path '{flatbuffer_path}'\")"
+      ],
+      "execution_count": 5,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "text": [
+            "Wrote .vmfb to path '/tmp/iree/colab_artifacts/counter_vmvx.vmfb'\n"
+          ],
+          "name": "stdout"
+        }
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "h8cmF6nAfza0",
+        "outputId": "85b91f2f-baec-4270-80f1-69742caf3f80"
+      },
+      "source": [
+        "#@title Test running the compiled VM module using IREE's runtime\n",
+        "\n",
+        "vm_module = ireert.VmModule.from_flatbuffer(flatbuffer_blob)\n",
+        "backend = module_utils.BackendInfo(\"iree_vmvx\")\n",
```

Comment:
In the future this will be optional for the runtime which will make repurposing the notebooks easier - there would only be one place that takes the config (the compiler) and that could be its own cell to talk about the backends/make it easier to switch. Specifying the target is an important part of the system and it's important to talk about it as part of the compilation flow - no one is going to want to use vmvx and having it easy to try out the other backends shows where that determination falls in the pipeline and where they would have to put it when building their own workflow.

Reply:
I at least dropped the dep on `module_utils` (which was just doing a dictionary remapping from `iree_vmvx` to `vmvx` :P).